Python 3.11 support

Hi @jameskermode , 

As conda-forge is moving forward in adding support for Python 3.11 we noticed that `f90wrap` is not yet compatible. If you have a moment it would be great to add Python 3.11 support. 
https://github.com/conda-forge/f90wrap-feedstock/pull/32

Best, 

Jan

It seems to be this part:
```
 f90wrap/arraydatamodule.c: In function 'PyInit_arraydata':
  f90wrap/arraydatamodule.c:171:28: error: lvalue required as left operand of assignment
    171 |   Py_TYPE(&PyFortran_Type) = &PyType_Type;
        |                            ^
```

--- a/.github/workflows/build-wheels.yml
+++ b/.github/workflows/build-wheels.yml
@@ -77,6 +77,11 @@ jobs:
                 else
                   for CIBW_MANYLINUX_X86_64_IMAGE in manylinux2010 manylinux2014; do
                     export CIBW_MANYLINUX_X86_64_IMAGE
+                    if [[ $CIBW_MANYLINUX_X86_64_IMAGE == manylinux2010 ]]; then
+                      export CIBW_SKIP="cp27-* cp35-* cp36-* cp311-* pp* *musllinux*"
+                    else
+                      export CIBW_SKIP="cp27-* cp35-* cp36-* pp* *musllinux*"
+                    fi
                     python -m cibuildwheel --output-dir wheelhouse --archs $ARCH
                   done
                 fi

--- a/.github/workflows/build-wheels.yml
+++ b/.github/workflows/build-wheels.yml
@@ -82,7 +82,7 @@ jobs:
                 fi
              done
         env:
-           CIBW_SKIP: cp27-* cp35-* cp36-* cp311-* pp* *musllinux*
+           CIBW_SKIP: cp27-* cp35-* cp36-* pp* *musllinux*
            CIBW_BEFORE_ALL_MACOS: "brew install gfortran && brew unlink gfortran && brew link gfortran"
           #  CIBW_BEFORE_ALL_LINUX: "which yum && yum install -y gcc-gfortran || apk add gfortran"
            CIBW_BEFORE_BUILD: "pip install oldest-supported-numpy"

--- a/.github/workflows/build-wheels.yml
+++ b/.github/workflows/build-wheels.yml
@@ -30,7 +30,7 @@ jobs:
           python-version: '3.10'
 
       - name: Install cibuildwheel
-        run: python -m pip install cibuildwheel==2.9.0
+        run: python -m pip install cibuildwheel==2.11.4
 
       - name: Build wheels
         run: |

--- a/.github/workflows/python-package.yml
+++ b/.github/workflows/python-package.yml
@@ -19,7 +19,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: [3.7, 3.8, 3.9, '3.10']
+        python-version: [3.7, 3.8, 3.9, '3.10', '3.11']
 
     steps:
     - uses: actions/checkout@v2

--- a/f90wrap/arraydatamodule.c
+++ b/f90wrap/arraydatamodule.c
@@ -168,7 +168,7 @@ PyMODINIT_FUNC
 PyInit_arraydata(void)
 {
   PyObject *mod = PyModule_Create(&arraydataModuleDef);
-#ifdef PY_MINOR_VERSION >= 11 // https://peps.python.org/pep-0674/#port-c-extensions-to-python-3-11
+#if PY_MINOR_VERSION >= 11 // https://peps.python.org/pep-0674/#port-c-extensions-to-python-3-11
   Py_SET_TYPE(&PyFortran_Type, &PyType_Type);
 #else
   Py_TYPE(&PyFortran_Type) = &PyType_Type;

--- a/f90wrap/arraydatamodule.c
+++ b/f90wrap/arraydatamodule.c
@@ -168,7 +168,11 @@ PyMODINIT_FUNC
 PyInit_arraydata(void)
 {
   PyObject *mod = PyModule_Create(&arraydataModuleDef);
+#ifdef PY_MINOR_VERSION >= 11 // https://peps.python.org/pep-0674/#port-c-extensions-to-python-3-11
+  Py_SET_TYPE(&PyFortran_Type, &PyType_Type);
+#else
   Py_TYPE(&PyFortran_Type) = &PyType_Type;
+#endif
 
   import_array();
   return mod;

